src/conf_mode/interfaces-openvpn.py

```diff
@@ -647,6 +648,13 @@ def apply(openvpn):
 
         return None
 
+    # verify specified IP address is present on any interface on this system
+    # Allow to bind service to nonlocal address, if it virtaual-vrrp address
+    # or if address will be assign later
+    if 'local_host' in openvpn:
+        if not is_addr_assigned(openvpn['local_host']):
+            cmd('sysctl -w net.ipv4.ip_nonlocal_bind=1')
```

Comment:
I wonder if we probably should make it a default option in https://github.com/vyos/vyos-1x/blob/current/src/etc/sysctl.d/30-vyos-router.conf

as the value is set here once but never unset.

Reply:
The value will be set only if it doesn’t find a binding address, in other cases it doesn’t touch this sysctl parameter. I’m not sure about default value, in theory it shouldn’t cause of reasons of other issues.

Reply:
The value is here set once, but never reset - so either we reset the value (which could probably trigger other funny behavior) or we can:

* Make OpenVPN systemd service file dependant on keepalived if both are configured so the startup order is ensured
* Always enable `net.ipv4.ip_nonlocal_bind=1` and probably for IPv6

Reply:
We can brake something else if we set it by default
https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt
```
ip_nonlocal_bind - BOOLEAN
	If set, allows processes to bind() to non-local IP addresses,
	which can be quite useful - but may break some applications.
	Default: 0
```

Reply:
> * Make OpenVPN systemd service file dependant on keepalived if both are configured so the startup order is ensured

The issue is not related exactly to vrrp, it can be PPPoE or IPSec which for some reason yet not established.

Reply:
I will change this to enable `ip_nonlocal_bind` for testin in `current` as discussed today